src/main/controller/WorkoutController.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from 'express';
+import db from '../models';
+import { Route } from '../constant/Route';
+
+export class WorkoutController {
+  // (9 - CC & WP) Retrieve detailed information about workout (title, duration, video link), exercises and their order, and info regarding exercises
+  getWorkoutDetails = async (req: Request, res: Response): Promise<any> => {
+    try {
+      if (!req.params.workoutID) {
+        throw `Missing workoutId param. usage: /${Route.WORKOUTS}/:workoutID(\\b[0-9a-f]{8}\\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\\b[0-9a-f]{12}\\b)`;
+      }
+
+      // TODO: make sure the user has permissions to view this workout (userId check? private / public check?)
+      const workouts = await db.Workout.findOne({
+        where: {
+          id: req.params.workoutID,
+        },
+        attributes: ['name', 'totalWorkoutTime', 'imageUrl'],
+        include: [
```

Comment:
Does this guarantee that exercises are returned in order? The task is  "Retrieve detailed information about [...] exercises and their order".

Reply:
This doesn't because we're missing `orderNum`- I was originally planning to add it by today but I'm behind schedule rn! I don't want to keep you blocked though @Haxrox  - do you want to just add it to this PR?